index.js

```diff
@@ -317,17 +318,21 @@ class Installer {
 
   updateInstallScript (dep, pkg) {
     const depPath = dep.path(this.prefix)
-    return statAsync(path.join(depPath, 'binding.gyp'))
-      .catch(err => { if (err.code !== 'ENOENT') { throw err } })
-      .then(stat => {
-        if (stat) {
-          if (!pkg.scripts) {
-            pkg.scripts = {}
-          }
-          pkg.scripts.install = 'node-gyp rebuild'
-        }
+    return checkPlatform(pkg, this.config.get('force'))
+      .then(() => {
+        return statAsync(path.join(depPath, 'binding.gyp'))
+          .catch(err => { if (err.code !== 'ENOENT') { throw err } })
+          .then(stat => {
+            if (stat) {
+              if (!pkg.scripts) {
+                pkg.scripts = {}
+              }
+              pkg.scripts.install = 'node-gyp rebuild'
+            }
+          })
+          .then(pkg)
       })
-      .then(pkg)
+      .catch(() => 'ignore')
```

Comment:
If you look at the `buildTree()` method, you'll notice a `if (!checkDepEnv(...)) {...}`. What that does is filter out the dependency altogether while building the tree we're going to extract. If you add the necessary checks to `checkDepEnv()`, you should be good.

Interesting to note that `checkPlatform()` is actually synchronous, even though it takes a callback: https://github.com/npm/npm-install-checks/blob/master/index.js#L27-L54 so I'd just rather you write a wrapper that treats it as sync so you don't have to refactor all the `checkDepEnv` calls into async calls.